### PR TITLE
Document for Maven research and some code modifications.

### PR DESCRIPTION
**Pull request by ClintonCao**
I did the following things:
- I wrote a research document for Maven (please refer to #79 for more information).
- Removed Octopull from Travis config.
- Updated config file for Better Code Hub.
- Modified code for visualizer:
 - All JSON files are written to the same folder.
 - Changed `ruleSet` to message in `PMDWarning`.
 - Added `message` field in the superclass `Warning`.
 - Extra variable is needed (project name) in the export of warnings for the visualizer.
 - Changed to a better variable name in `GUIController`.

Please review and if there are any questions, let me know :)

### CODE REVIEW

**Comment by ClintonCao** [comment]
I've also added the following things to the README:
- link for Maven Research document (link does not exist yet).
- Requirement for the program is added under "Requirements".

File: src/main/java/BlueTurtle/warnings/PMDWarning.java [hunk 1]
```diff
@@ -62,25 +60,25 @@ public boolean equals(Object other) {
 
 		return (filePath.equals(that.filePath) && fileName.equals(that.fileName) && line == that.line
 				&& classification.equals(that.classification) && packageName.equals(that.packageName) && type.equals(that.type)
```
Comment (SunweiWang) [inline src/main/java/BlueTurtle/warnings/PMDWarning.java]: In PMD, there are no default messages, and ruleSet is used to concatenate with ruleName to look up in the gdc, so I will change this back.

**Comment by ClintonCao** [comment]
Ahh must have confused it with something..

**Comment by Michieldoesburg** [comment]
Good stuff, I think we can merge this.

**Comment by SunweiWang** [comment]
I would say merge this, but I am also contributing to this :D

**Comment by SunweiWang** [comment]
Well, since we have already majority of the team reviewing this branch and the build passes, I'm going to merge this PR.

**Pull request merged**